client/src/driver/BitbakeRecipeScanner.ts

```diff
@@ -0,0 +1,74 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+import { type LanguageClient } from 'vscode-languageclient/node'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { runBitbakeTask } from '../ui/BitbakeCommands'
+import { type BitbakeCustomExecution, type BitbakeTaskProvider } from '../ui/BitbakeTaskProvider'
+import { RequestMethod, type RequestParams } from '../lib/src/types/requests'
+
+export class BitbakeRecipeScanner {
+  private isPending: boolean = false
+  private _languageClient: LanguageClient | undefined
+  private _currentUriForScan: string = ''
+
+  public scanResults: string = ''
+  public processedScanResults: Record<string, unknown> | undefined
+
+  async scan (chosenRecipe: string, taskProvider: BitbakeTaskProvider, uri: any): Promise<void> {
+    logger.debug(`[BitbakeRecipeScanner] Scanning recipe env: ${uri}`)
+    this._currentUriForScan = uri
+
+    const scanRecipeEnvTask = new vscode.Task(
+      { type: 'bitbake', recipes: [chosenRecipe], options: { parseOnly: true, env: true } },
+      vscode.TaskScope.Workspace,
+      'Bitbake: Scan recipe env',
+      'bitbake'
+    )
+
+    const runningTasks = vscode.tasks.taskExecutions
+    if (runningTasks.some((execution) => execution.task.name === scanRecipeEnvTask.name)) {
+      logger.debug('[BitbakeRecipeScanner] Recipe scan is already running')
+      this.isPending = true
```

Comment:
If the pending scan is for another recipe than the one being scaned, you'll not scan the new one. I suggest that isPending stores the chosenRecipe name to scan next time. The variable could be renamed `this.scanPendingRecipe` which would be `string | undefined`

Reply:
Twisted code a little bit more to allow more than 1 pending recipe. My goal is to execute them in a queue-like manner. Check ee0019c4f8a25e6b80d80675724c3a5da45063b0

Reply:
I don't think queuing more than one is a good idea. Parsing takes a few seconds. If a user does many edits at the same time, bitbake can be unavailable for minutes! for example multiple find/replaces then saves. Just queuing one is enough.